internal/plugin/plugin.go

```diff
@@ -93,12 +95,39 @@ func NewPluginClient(connector string, version string, logger hclog.Logger) (Plu
 		return nil, fmt.Errorf("the plugin (%s) doesn't correctly implement the necessary interfaces", connector)
 	}
 
+	currentVersion := version.CurrentCliVersionInformation{}
+	if _, err = isPluginCompatibleWithCli(&currentVersion, is, logger); err != nil {
+		logger.Error("Plugin not compatible: %s", err.Error())
```

Comment:
We'll need to do some good explaining here to the user on how to solve the problem.